src/uart/mod.rs

```diff
@@ -96,27 +172,39 @@ impl<Port: private::UARTPortCompatable> UART<Port> {
         uart.clear_rx_fifo();
         uart.clear_tx_fifo();
 
-        // Set the character length to 8
-        uart.set_character_length(CharacterLength::EightBits);
-
-        // Set the number of stop bits to 1
-        uart.set_number_stop_bits(StopBits::OneBit);
+        unsafe {
+            uart.reg.set_character_length(character_length as u8);
+            // Set the number of stop bits to 1
+            let stop_bits = match stop_bits {
+                StopBits::OneBit => true,
+                StopBits::TwoBits => false,
+            };
+            uart.reg.set_number_of_stop_bits(stop_bits);
+            uart.reg
+                .set_transmit_parity_generation_enable(transmit_parity);
+            // Set the parity value
+            let parity_value = match parity_value {
+                ParityValueSelect::OneBased => false,
+                ParityValueSelect::ZeroBased => true,
+            };
+            uart.reg.set_parity_value(parity_value);
+            // Set the clock source to IBRO
+            uart.reg.set_baud_clock_source(ClockSources::IBRO as u8);
+            // Set the clock divisor to 7.3728 Mhz / baud rate
+            let divisor = 7372800 / baud_rate as u32;
+            uart.reg.set_baud_rate_divisor(divisor);
+            // Set the Hardware Flow Control
+            uart.reg.set_hardware_flow_control(hfc);
+        }
 
-        // Dissable parity
-        uart.transmit_parity_enable(false);
-        // Set the Divisor to 64
-        uart.set_clock_divisor(64);
-        // Set the clock source
-        uart.set_baud_clock_source(ClockSources::IBRO);
         uart
     }
 
     /// # Print String
     /// Prints the string passed
```

Comment:
Include in your comment that this function will call `write_blocking_transmit_fifo`. We want to know as a caller that this function is also blocking.